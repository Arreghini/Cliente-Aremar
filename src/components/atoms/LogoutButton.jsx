import React from 'react';
import { useAuth0 } from '@auth0/auth0-react';

const LogoutButton = () => {
  const { logout } = useAuth0();

  return (
    <button
      onClick={() =>
        logout({ logoutParams: { returnTo: window.location.origin } })
      }
      type="submit"
      className="px-0 py-2 w-full text-left text-white hover:text-yellow-500 cursor-pointer"
    >
      Logout
    </button>
  );
};

export default LogoutButton;
