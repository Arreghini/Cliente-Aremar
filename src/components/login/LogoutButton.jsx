import React from 'react';
import { useAuth0 } from '@auth0/auth0-react';

const logoutButton = () => {
    const { logout } = useAuth0();

return (
    <div className="flex justif y-center justify-center mt-4">
    <button onClick={() => logout({ logoutParams: { returnTo: window.location.origin } })}
    type="submit"
    className="w-1/3 justify-center bg-yellow-500 text-white font-semibold text-xl py-2 px-4 rounded hover:bg-yellow-600"
    >
    LOGOUT
    </button>
    </div>
    )
}

export default logoutButton;


