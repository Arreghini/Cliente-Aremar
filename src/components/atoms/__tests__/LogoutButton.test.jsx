// src/components/atoms/__tests__/LogoutButton.test.jsx
import React from 'react';
import { describe, it, expect, vi, beforeEach } from 'vitest';
import { render, screen, fireEvent } from '@testing-library/react';
import LogoutButton from '../LogoutButton';
import { useAuth0 } from '@auth0/auth0-react';

// Mock del hook de Auth0
vi.mock('@auth0/auth0-react', () => ({
  useAuth0: vi.fn(),
}));

describe('LogoutButton', () => {
  let logoutMock;

  beforeEach(() => {
    logoutMock = vi.fn();
    useAuth0.mockReturnValue({
      logout: logoutMock,
    });
  });

  it('renderiza el botón correctamente', () => {
    render(<LogoutButton />);
    const button = screen.getByRole('button', { name: /logout/i });
    expect(button).toBeInTheDocument();
  });

  it('el botón tiene el texto correcto', () => {
    render(<LogoutButton />);
    const button = screen.getByRole('button');
    expect(button).toHaveTextContent('Logout');
  });

  it('el botón tiene clase y tipo correctos', () => {
    render(<LogoutButton />);
    const button = screen.getByRole('button');
    expect(button).toHaveClass(
      'px-0 py-2 w-full text-left text-white hover:text-yellow-500 cursor-pointer'
    );
    expect(button).toHaveAttribute('type', 'submit');
  });

  it('llama a logout al hacer clic con returnTo correcto', () => {
    render(<LogoutButton />);
    const button = screen.getByRole('button');
    fireEvent.click(button);
    expect(logoutMock).toHaveBeenCalledTimes(1);
    expect(logoutMock).toHaveBeenCalledWith({
      logoutParams: { returnTo: window.location.origin },
    });
  });

  it('coincide con el snapshot', () => {
    const { container } = render(<LogoutButton />);
    expect(container).toMatchSnapshot();
  });
});
