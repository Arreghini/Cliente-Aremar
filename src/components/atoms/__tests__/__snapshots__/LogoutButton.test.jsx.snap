// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`LoginButton > coincide con el snapshot 1`] = `
<div>
  <button
    class="px-0 py-2 w-full text-left text-yellow-200 hover:text-yellow-500 cursor-pointer"
    type="submit"
  >
    Login
  </button>
</div>
`;

exports[`LogoutButton > coincide con el snapshot 1`] = `
<div>
  <button
    class="px-0 py-2 w-full text-left text-white hover:text-yellow-500 cursor-pointer"
    type="submit"
  >
    Logout
  </button>
</div>
`;
